common/dynamicconfig/constants.go

```diff
@@ -24,1049 +24,2292 @@
 
 package dynamicconfig
 
-const (
+import (
+	"os"
+	"time"
+
+	enumspb "go.temporal.io/api/enums/v1"
+
+	"go.temporal.io/server/common/primitives"
+	"go.temporal.io/server/common/retrypolicy"
+)
+
+var (
 	// keys for admin
 
-	// AdminEnableListHistoryTasks is the key for enabling listing history tasks
-	AdminEnableListHistoryTasks = "admin.enableListHistoryTasks"
-	// AdminMatchingNamespaceToPartitionDispatchRate is the max qps of any task queue partition for a given namespace
-	AdminMatchingNamespaceToPartitionDispatchRate = "admin.matchingNamespaceToPartitionDispatchRate"
-	// AdminMatchingNamespaceTaskqueueToPartitionDispatchRate is the max qps of a task queue partition for a given namespace & task queue
-	AdminMatchingNamespaceTaskqueueToPartitionDispatchRate = "admin.matchingNamespaceTaskqueueToPartitionDispatchRate"
+	AdminEnableListHistoryTasks = NewBoolGlobalSetting(
+		"admin.enableListHistoryTasks",
```

Comment:
Sure, @tdeebswihart mentioned that too. There's no reason it has to be in one file (it doesn't have to be now either), we can e.g. put the matching stuff in `service/matching/dynamicconfig.go`. But maybe let's move things around in follow-up PRs?

Reply:
What do you mean by sub-namespaces and what functionality do you want from that?

Reply:
Let's do these changes in a follow-up, this PR is large enough as is

Reply:
> What do you mean by sub-namespaces and what functionality do you want from that?

Just meant multiple dots: `foo.bar.zzz`, `a.b.c.d`, etc.
Not an additional functionality, just felt that most (all?) of our dynamic configs have only one "prefix" (eg: `system`, `frontend`, etc), but maybe we should start introducing more (eg: `system.visibility.enableManualPagination` instead of `system.visibilityEnableManualPagination`)

Reply:
We can totally do that. The dot has no special meaning, they're just strings.